src/wp-includes/class-wp-theme-json-data.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * WP_Theme_JSON_Data class
+ *
+ * @package WordPress
+ * @subpackage Theme
+ * @since 6.1.0
+ */
+
+/**
+ * Class to provide access to update a theme.json structure.
+ */
+class WP_Theme_JSON_Data {
+
+	/**
+	 * Container of the data to update.
+	 *
+	 * @since 6.1.0
+	 * @var WP_Theme_JSON
+	 */
+	private $theme_json = null;
+
+	/**
+	 * The origin of the data: default, theme, user, etc.
+	 *
+	 * @since 6.1.0
+	 * @var string
+	 */
+	private $origin = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array  $data   Array following the theme.json specification.
```

Comment:
It would be great to add a reference to the documentation that explains the data format used with `theme.json`.